client/src/analytics_component.rs

```diff
@@ -1,68 +1,520 @@
+            </td>
+            <td class="country-cell">
+                <span class="country-code">{ &analytics.iso_code }</span>
+            </td>
+            <td class="count-cell">{ format_count(&(analytics.count as i32)) }</td>
         </tr>
     }
 }
 
+#[function_component(LoadingSpinner)]
+fn loading_spinner() -> Html {
+    html! {
+        <div class="loading-container">
+            <div class="loading-spinner"></div>
+            <span class="loading-text">{"Loading analytics..."}</span>
+        </div>
+    }
+}
+
+#[derive(Properties, PartialEq)]
+pub struct ToolbarProps {
+    on_filter_change: Callback<String>,
+    on_search_change: Callback<String>,
+    on_view_change: Callback<ViewType>,
+    on_export: Callback<()>,
+    view_type: ViewType,
+    data: Vec<Analytics>,
+}
+
+#[derive(PartialEq, Clone)]
+pub enum ViewType {
+    Table,
+    Globe,
+}
+
+#[derive(PartialEq, Clone)]
+pub enum SortField {
+    IpAddress,
+    Path,
+    Country,
+    Visits,
+}
+
+#[derive(PartialEq, Clone)]
+pub enum SortDirection {
+    Ascending,
+    Descending,
+}
+
+fn sort_analytics(data: &mut [Analytics], field: &SortField, direction: &SortDirection) {
+    data.sort_by(|a, b| {
+        let ordering = match field {
+            SortField::IpAddress => a.ip_address.cmp(&b.ip_address),
+            SortField::Path => a.path.cmp(&b.path),
+            SortField::Country => a.iso_code.cmp(&b.iso_code),
+            SortField::Visits => a.count.cmp(&b.count),
+        };
+        match direction {
+            SortDirection::Ascending => ordering,
+            SortDirection::Descending => ordering.reverse(),
+        }
+    });
+}
+
+#[function_component(AnalyticsToolbar)]
+fn analytics_toolbar(props: &ToolbarProps) -> Html {
+    html! {
+        <div class="tools-section">
+            <div class="search-filter-group">
+                <div class="search-group">
+                    <label>{"Search:"}</label>
+                    <input
+                        type="text"
+                        placeholder="Filter by IP or path..."
+                        onchange={props.on_search_change.reform(|e: Event| {
+                            let input = e.target().unwrap().unchecked_into::<HtmlInputElement>();
+                            input.value()
+                        })}
+                    />
+                </div>
+                <div class="filter-group">
+                    <label>{"Country:"}</label>
+                    <select onchange={props.on_filter_change.reform(|e: Event| {
+                        let select = e.target().unwrap().unchecked_into::<HtmlSelectElement>();
+                        select.value()
+                    })}>
+                        <option value="">{"All Countries"}</option>
+                        <option value="US">{"United States"}</option>
+                        <option value="GB">{"United Kingdom"}</option>
+                        // Add more countries as needed
```

Comment:
There are many more countries down below. Why is "GB" "United Kingdom" and not "UK"?